src/stream/src/executor/lookup/impl_.rs

```diff
@@ -393,46 +396,28 @@ impl<S: StateStore> LookupExecutor<S> {
             return Ok(result.iter().cloned().collect_vec());
         }
 
-        // Serialize join key to a state store key.
-        let key_prefix = {
-            let mut key_prefix = vec![];
-            // Manually encode a vnode here to match the encoding of cell-based table
-            // TODO: refactor lookup with cell-based table
-            key_prefix.extend(DEFAULT_VNODE.to_be_bytes());
-            self.arrangement
-                .serializer
-                .serialize_datums(lookup_row.0.iter(), &mut key_prefix);
-            key_prefix
-        };
-
-        tracing::trace!(target: "events::stream::lookup::lookup_row", "{:?}, {:?}", lookup_row, bytes::Bytes::copy_from_slice(&key_prefix));
-
-        let arrange_keyspace = self.arrangement.keyspace.append(key_prefix);
-        let all_cells = arrange_keyspace.scan(None, lookup_epoch).await?;
+        tracing::trace!(target: "events::stream::lookup::lookup_row", "{:?}", lookup_row);
 
         let mut all_rows = vec![];
-
-        // The key is truncated in `arrange_keyspace` so there's no vnode to decode
-        // TODO: refactor lookup with cell-based table and remove `deserialize_with_vnode`
-        for (pk_with_cell_id, cell) in all_cells {
-            tracing::trace!(target: "events::stream::lookup::scan", "{:?} => {:?}", pk_with_cell_id, cell);
-            if let Some((_, _, row)) = self
-                .arrangement
-                .deserializer
-                .deserialize_without_vnode(&pk_with_cell_id, &cell)?
-            {
-                all_rows.push(row);
-            }
-        }
-
-        if let Some((_, _, last_row)) = self.arrangement.deserializer.take() {
-            all_rows.push(last_row);
+        let all_data_iter = self
+            .arrangement
+            .state_table
+            .iter_with_pk_prefix(&lookup_row, lookup_epoch)
+            .await?
+            .fuse();
+        pin_mut!(all_data_iter);
+        while let Some(inner) = all_data_iter.next().await {
+            let ret = inner?;
+            all_rows.push(ret.into_owned());
         }
```

Comment:
```suggestion
     {
        let all_data_iter = self
            .arrangement
            .state_table
            .iter_with_pk_prefix(&lookup_row, lookup_epoch)
            .await?
            .fuse();
        pin_mut!(all_data_iter);
        while let Some(inner) = all_data_iter.next().await {
            let ret = inner?;
            all_rows.push(ret.into_owned());
        }
     }
```

Perhaps you can use scope to show compiler that `all_data_iter` is not used anywhere else.
Seems to work for me on my end.

Reply:
work pretty well